### PR TITLE
Integrate TinyDB and CRUD routes.

### PR DESCRIPTION
**Pull request by Youngestdev**
# Goal

The TinyDB database is used to add persistence and enable the API to track changes made to the policy stored as a JSON file.

## Changes

- `database/policy_database.py`: This file holds the CRUD functions required for interaction from the API.

- I updated the existing route: `POST` route to interact with the tinydb database.

I also implemented:

- The `UPDATE` route is responsible for updating the policies in the database as well as the repository. This operation is done in parallel.

- I implemented the `GET` route for retrieving single policies. This will come in handy when building the frontend.

- I implemented the `DELETE` route for removing policies from the database.

### CODE REVIEW

File: app/database/policy_database.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from fastapi import HTTPException
+from tinydb import Query, TinyDB
+from app.schemas.rules import RequestObject
+
+database = TinyDB('/tmp/fastgeoapi/database.json')
```
Comment (francbartoli) [inline app/database/policy_database.py:5]: could this be a configuration from `.env` file?

File: app/server/api.py [hunk 2]
```diff
@@ -1,19 +1,57 @@
...
+from app.database.policy_database import add_policy, update_policy, exists, get_policy, delete_policy
+from app.schemas.rules import RequestObject, UpdateRequestObject
 from app.utils.write_rego import write_to_file
 
 app = FastAPI()
 
 
-@app.post("/save")
-async def write(rego_rule: RequestObject) -> dict:
+@app.post("/policy/save")
```
Comment (francbartoli) [inline app/server/api.py:10]: Having only `/policies` would be more restful

File: app/server/api.py [hunk 3]
```diff
@@ -1,19 +1,57 @@
...
     response = write_to_file(rego_rule)
 
     response["state"] = rego_rule
     return response
 
 
-@app.put("/save")
-async def modify(rego_rule: RequestObject):
-    pass
+@app.get("/policy/{name}")
```
Comment (francbartoli) [inline app/server/api.py:19]: why not `{policy}` rather than `{name}` like below?

File: app/server/api.py (hunk 2 above)
Comment (Youngestdev) [reply app/server/api.py:10]: Okay, cool.

**Pull request merged**